jpype/_jproxy.py

```diff
@@ -22,46 +22,64 @@
 # FIXME the java.lang.method we are overriding should be passes to the lookup function
 # so we can properly handle name mangling on the override.
 
-
-def _createJProxy(cls, *intf, **kwargs):
-    """ (internal) Create a proxy from a python class with
-    @JOverride notation on methods.
-    """
-    # Convert the interfaces list
-    actualIntf = _convertInterfaces(intf)
-
-    # Find all class defined overrides
-    overrides = {}
-    for k, v in cls.__dict__.items():
-        try:
-            attr = object.__getattribute__(v, "__joverride__")
-            overrides[k] = (v, attr)
-        except AttributeError:
-            pass
-
+def _checkInterfaceOverrides(interfaces, overrides):
     # Verify all methods are overriden
-    for interface in actualIntf:
+    for interface in interfaces:
         for method in interface.class_.getMethods():
             if method.getModifiers() & 1024 == 0:
                 continue
             if not str(method.getName()) in overrides:
                 raise NotImplementedError("Interface '%s' requires method '%s' to be implemented." % (
                     interface.class_.getName(), method.getName()))
 
-    members = {}
+
+def _createJProxy(cls, *intf, **kwargs):
+    """ (internal) Create a proxy from a python class with
+    @JOverride notation on methods.
+    """
+    # TODO: When Python2 is dropped these kwargs can become keyword-only kwargs
```

Comment:
Is this already the case? https://github.com/jpype-project/jpype/commit/43eae033964555de882b807d04c2f0b74465c374#r38120855

Reply:
Yes python 2 is dropped but 3.5 is still in scope so we have to limit to 3.5 features